src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobFileStore.cs

```diff
@@ -438,6 +445,14 @@ protected virtual BlockBlobClient GetInstanceBlockBlobClient(long version, strin
         return _blobClient.BlobContainerClient.GetBlockBlobClient(fullPath);
     }
 
+    /// <summary>
+    /// Get client based on watermark/version and partition when using internal store and based on file properties
+    /// when using external store.
+    /// </summary>
+    /// <param name="version">Version of file to get</param>
+    /// <param name="partition">Partition within which the file should live in</param>
+    /// <param name="fileProperties">File properties to use for external store. If not using external store, set to null.</param>
+    /// <remarks>Always use on existing files whether using for external or internal store.</remarks>
```

Comment:
Do you think can we simplify into one method? We can return the client based on fileProperties being null or not. Some one may use wrong method.

Reply:
I get that, but I still don't understand why we can't have in one method. 

```
protected virtual BlockBlobClient GetExistingInstanceBlockBlobClient(long version, Partition partition, FileProperties fileProperties)
{
	EnsureArg.IsNotNull(partition, nameof(partition));
	if (_blobClient.IsExternal && fileProperties != null)
	{
		// does not throw, just appends uri with blobName
		return _blobClient.BlobContainerClient.GetBlockBlobClient(fileProperties.Path);
	}
	string blobName = _nameWithPrefix.GetInstanceFileName(version);
	string fullPath = _blobClient.GetServiceStorePath(partition.Name) + blobName;
	// does not throw, just appends uri with blobName
	return _blobClient.BlobContainerClient.GetBlockBlobClient(fullPath);
}
```

Reply:
I am fine keeping it separate, it is just it may lead to wrong usage.